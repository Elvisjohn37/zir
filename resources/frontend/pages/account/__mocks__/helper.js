let mockedModule = jest.createMockFromModule('../helper.js');

mockedModule.openAccountWindow = jest.fn();

module.exports = mockedModule;
