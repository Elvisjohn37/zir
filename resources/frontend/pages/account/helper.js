export function openAccountWindow(url) {
	window.open(url, 'account_window', 'width = 670, height = 800');
}
