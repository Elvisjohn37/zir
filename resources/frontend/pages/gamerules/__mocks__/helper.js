let mockedModule = jest.createMockFromModule('../helper');

mockedModule.extractProductGameType = (gameRulesList) => {
	let extractedItems = [];
	let indexMap = {};
	let nextIndex = 0;

	gameRulesList.forEach((item) => {
		if (item.productName in indexMap) {
			if (!extractedItems[indexMap[item.productName]].gameTypes.includes(item.gameTypeName)) {
				extractedItems[indexMap[item.productName]].gameTypes.push(item.gameTypeName);
			}
		} else {
			indexMap[item.productName] = nextIndex;
			nextIndex++;

			extractedItems.push({
				productName: item.productName,
				gameTypes: [item.gameTypeName],
			});
		}
	});

	return extractedItems;
};

module.exports = mockedModule;
