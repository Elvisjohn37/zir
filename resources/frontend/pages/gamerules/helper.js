import lodash from 'lodash';

function extractProductGameType(gameRulesList) {
	let extractedItems = [];
	let indexMap = {};
	let nextIndex = 0;

	gameRulesList.forEach((item) => {
		if (item.productName in indexMap) {
			if (!lodash.includes(extractedItems[indexMap[item.productName]].gameTypes, item.gameTypeName)) {
				extractedItems[indexMap[item.productName]].gameTypes.push(item.gameTypeName);
			}
		} else {
			indexMap[item.productName] = nextIndex;
			nextIndex++;

			extractedItems.push({
				productName: item.productName,
				gameTypes: [item.gameTypeName],
			});
		}
	});

	return extractedItems;
}

export { extractProductGameType };
