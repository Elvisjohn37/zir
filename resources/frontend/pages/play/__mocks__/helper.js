let postMessageHandler = () => jest.fn();

let setPostMessageHandler = () => jest.fn();

let resetPostMessageHandler = () => jest.fn();

resetPostMessageHandler();

// _________________________________________

let windowResponseData,
	isMobile = false;

let windowResponse = () => windowResponseData;

let resetWindowResponse = () => {
	windowResponseData = {
		result: true,
		data: {
			url: 'https://eyecon.classicku.com/launch/lek?uid=ANqWb0TXAFr1%252BZT7aQmY%252F1M%253D&guid=124275588462ff2e5f3973e3.1098258&gameid=71656&brand=LEK&cur=ZWD&lang=en&homeURL=https%3A%2F%2Fgames.classicku.com%2Fclosegame%3FgameID%3DDVbr%26clientID%3DML641',
			title: 'Beat The Bobbies',
			maxPayout: '700000000.000000',
			isMobile: false,
			viewport: 'desktop',
			productID: 'zy',
			balanceInfo: { playableBalance: '1457.800000', currencyCode: 'ZWD' },
		},
	};
};

let setWindowResponse = (data) => {
	windowResponseData = {
		...windowResponseData,
		data: {
			...windowResponseData.data,
			...data,
		},
	};
};

let setErrorResponse = () => {
	windowResponseData = {
		...windowResponseData,
		result: false,
	};
};

resetWindowResponse();

// _________________________________________

module.exports = { postMessageHandler, windowResponse, setWindowResponse, resetWindowResponse, setErrorResponse };
