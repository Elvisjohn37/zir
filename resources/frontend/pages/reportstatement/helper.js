import { getStatementReport as getStatementReportAjax } from 'frontend/ajax/backend';

export function getStatementReport(
	isMounted,
	setDateRange,
	statementRange,
	setStatementData,
	setStatementDataLoading,
	setStatementDataError
) {
	setStatementDataLoading(true);

	getStatementReportAjax(
		statementRange,
		!isMounted,
		(response) => {
			setStatementDataLoading(false);
			setStatementData({
				content: response.data.data.content,
				footer: response.data.data.footer,
			});
			if (!isMounted) {
				setDateRange({
					isLoading: false,
					range: response.data.data.dateRange,
				});
			}
		},
		() => {
			setStatementDataError(true);
			setStatementDataLoading(false);
		}
	);
}
