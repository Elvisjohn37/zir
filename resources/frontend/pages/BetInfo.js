import React from 'react';
import { windowResponse } from 'frontend/utils/helper';
import { ErrorPage } from 'frontend/components/OnPageError';
import { useParams } from 'react-router-dom';
import BetInfoIframe from './betinfo/BetInfoIframe';
import lodash from 'lodash';

const BET_INFO_IFRAME = ['eyecon', 'funky'];

export default function BetInfo() {
	let response = windowResponse();
	let { type } = useParams();

	return response.result === true && lodash.includes(BET_INFO_IFRAME, type) ? (
		<BetInfoIframe type={type} />
	) : (
		<ErrorPage />
	);
}
