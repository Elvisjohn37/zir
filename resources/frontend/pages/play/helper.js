import { windowResponse } from 'frontend/utils/helper';

let postMessageHandlerWrapper = () => {
	let allowedOrigin = windowResponse() ? windowResponse().data.url : '';

	return function ({ data, origin }) {
		let originUrl = new URL(origin);
		let gameUrl = new URL(allowedOrigin);

		if ((gameUrl.origin == originUrl.origin || window.location.host == originUrl.host) && data == 'closeGame') {
			window.location.href = '/';
		}
	};
};

let postMessageHandler = postMessageHandlerWrapper();

export { postMessageHandler, windowResponse };
