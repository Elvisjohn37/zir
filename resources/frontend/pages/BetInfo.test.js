import React from 'react';
import BetInfo from './BetInfo';
import { render } from '@testing-library/react';
import { setWindowResponse, resetWindowResponse } from 'frontend/utils/helper';
import { setUseParams, resetUseParams } from 'react-router-dom';

jest.mock('frontend/utils/helper');
jest.mock('frontend/components/OnPageError');
jest.mock('react-router-dom');
jest.mock('./betinfo/BetInfoIframe');
jest.mock('lodash');

describe('</BetInfo>', () => {
	beforeEach(() => {
		resetWindowResponse();
		resetUseParams();
	});

	it('should match snapshot', () => {
		setWindowResponse({ result: true });
		setUseParams({ type: 'eyecon' });
		let { asFragment } = render(<BetInfo />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should load eyecon bet details', () => {
		setWindowResponse({ result: true });
		setUseParams({ type: 'eyecon' });
		let { getByTestId, queryByTestId } = render(<BetInfo />);
		let betInfoIframe = getByTestId('BetInfoIframe');
		let errorPage = queryByTestId('ErrorPage');

		expect(betInfoIframe).toBeInTheDocument();
		expect(betInfoIframe).toHaveAttribute('type', 'eyecon');
		expect(errorPage).not.toBeInTheDocument(0);
	});

	it('should load funky bet details', () => {
		setWindowResponse({ result: true });
		setUseParams({ type: 'funky' });
		let { getByTestId, queryByTestId } = render(<BetInfo />);
		let betInfoIframe = getByTestId('BetInfoIframe');
		let errorPage = queryByTestId('ErrorPage');

		expect(betInfoIframe).toBeInTheDocument();
		expect(betInfoIframe).toHaveAttribute('type', 'funky');
		expect(errorPage).not.toBeInTheDocument(0);
	});

	it('should display error page when the bet details type is unknown', () => {
		setWindowResponse({ result: true });
		setUseParams({ type: 'unknownType' });
		let { getByTestId } = render(<BetInfo />);

		expect(getByTestId('ErrorPage')).toBeInTheDocument();
	});

	it('should display error page when fetching of bet details in backend fails', () => {
		setWindowResponse({ result: false });
		setUseParams({ type: 'eyecon' });
		let { getByTestId } = render(<BetInfo />);

		expect(getByTestId('ErrorPage')).toBeInTheDocument();
	});
});
