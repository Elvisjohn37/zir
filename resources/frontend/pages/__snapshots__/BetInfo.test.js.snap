// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</BetInfo> should match snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="BetInfoIframe"
    type="eyecon"
  />
</DocumentFragment>
`;
